docs/src/main/paradox/client.md

```diff
@@ -156,7 +156,81 @@ Maven
 mvn akka-grpc:generate compile exec:java -Dexec.mainClass=io.grpc.examples.helloworld.GreeterClient
 ```
 
+### Configuration 
 
+A gRPC client is configured with a `GrpcClientSettings` instance. There are a number of ways of creating one and the API
+docs are the best reference. An `ActorSystem` is always required as it is used for default configuration and service discovery.
+
+The simplest way to create a client is to provide a static host and port.
+
+Scala
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/scala/docs/akka/grpc/client/GrpcClientSettingsCompileOnly.scala) { #simple }
+
+Java
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/java/jdocs/akka/grpc/client/GrpcClientSettingsCompileOnly.java) { #simple }
+
+Further settings can be added via the `with` methods
+
+Scala
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/scala/docs/akka/grpc/client/GrpcClientSettingsCompileOnly.scala) { #simple-programmatic }
+
+Java
+:  @@snip [GrpcClientSettingsCompileOnly]($root$/../runtime/src/test/java/jdocs/akka/grpc/client/GrpcClientSettingsCompileOnly.java) { #simple-programmatic }
+
+Instead a client can be defined in configuration. All client configurations need to be under `akka.grpc.client`
+
+Scala
+:  @@snip [GrpcClientSettingsSpec]($root$/../runtime/src/test/scala/akka/grpc/GrpcClientSettingsSpec.scala) { #client-config }
+
+Java
+:  @@snip [GrpcClientSettingsSpec]($root$/../runtime/src/test/scala/akka/grpc/GrpcClientSettingsSpec.scala) { #client-config }
+
+Clients defined in configuration pick up defaults from `reference.conf`:
+
+Scala
+:  @@snip [reference]($root$/../runtime/src/main/resources/reference.conf) { #defaults }
+
+Java
+:  @@snip [reference]($root$/../runtime/src/main/resources/reference.conf) { #defaults }
+
+#### Service discovery
+
+The examples above all use a hard coded host and port for the location of the gRPC service which is the default if you do not configure a `service-discovery-mechanism`.
+Alternatively [Akka Service Discovery](https://developer.lightbend.com/docs/akka-management/current/discovery.html) can be used.
```

Comment:
hm technically we call it Akka Discovery, perhaps good to keep that name for now, in case we grep rename later on?

Reply:
Hrmm we seem to use Service Discovery and Akka Discovery in equal measure e.g. https://developer.lightbend.com/docs/akka-management/current/discovery.html

I'll go with Akka Discovery and make note to update Akka Management docs

Reply:
Okey, sounds good :) thanks for checking!